mongodbatlas/resource_mongodbatlas_project_ip_access_list.go

```diff
@@ -278,3 +283,54 @@ func resourceMongoDBAtlasIPAccessListImportState(ctx context.Context, d *schema.
 
 	return []*schema.ResourceData{d}, nil
 }
+
+func isProjectIPAccessInTheList(ctx context.Context, conn *matlas.Client, projectID, entry string) (bool, error) {
+	currentPage := 1
+	exists := false
+	err := resource.RetryContext(ctx, 2*time.Minute, func() *resource.RetryError {
+		accessList, resp, err := conn.ProjectIPAccessList.List(ctx, projectID, &matlas.ListOptions{PageNum: currentPage})
+		if err != nil {
+			switch {
+			case strings.Contains(fmt.Sprint(err), "500"):
+				return resource.RetryableError(err)
+			case strings.Contains(fmt.Sprint(err), "404"):
+				return resource.RetryableError(err)
+			default:
+				return resource.NonRetryableError(fmt.Errorf(errorAccessListRead, err))
+			}
+		}
+
+		if accessList.TotalCount > 0 {
+			for _, result := range accessList.Results {
+				if result.IPAddress == entry || result.CIDRBlock == entry {
+					exists = true
+					break
+				}
+			}
+		}
+
```

Comment:
[nit] I think that this may be more readable 

```golang
		if exists {
			return nil
		}

	
		currentPage, err = resp.CurrentPage()
		if err != nil {
			return resource.NonRetryableError(fmt.Errorf(errorAccessListRead, err))
		}

		if !resp.IsLastPage() {
			currentPage++
			return resource.RetryableError(fmt.Errorf("[DEBUG] Current page : %d Next page: %d, will retry again", currentPage-1, currentPage))
		}
		
		return nil
```